test/common/conn_pool/conn_pool_base_test.cc

```diff
@@ -138,5 +139,33 @@ TEST_F(ConnPoolImplBaseTest, NoPrefetchIfDegraded) {
   pool_.destructAllConnections();
 }
 
+TEST_F(ConnPoolImplBaseTest, ExplicitPrefetch) {
+  // Create more than one connection per new stream.
+  ON_CALL(*cluster_, perUpstreamPrefetchRatio).WillByDefault(Return(1));
+  EXPECT_CALL(pool_, instantiateActiveClient).Times(AnyNumber());
+
+  // With global prefetch off, we won't prefetch.
+  EXPECT_FALSE(pool_.maybePrefetch(0));
+  // With prefetch ratio of 1.1, we'll prefetch two connections.
+  // Currently, no number of subsequent calls to prefetch will increase that.
+  EXPECT_TRUE(pool_.maybePrefetch(1.1));
+  EXPECT_TRUE(pool_.maybePrefetch(1.1));
+  EXPECT_FALSE(pool_.maybePrefetch(1.1));
+
+  // With a higher prefetch ratio, more connections may be prefetched.
+  EXPECT_TRUE(pool_.maybePrefetch(3));
+
+  pool_.destructAllConnections();
+}
+
+TEST_F(ConnPoolImplBaseTest, ExplicitPrefetchNotHealthy) {
+  // Create more than one connection per new stream.
+  ON_CALL(*cluster_, perUpstreamPrefetchRatio).WillByDefault(Return(1));
+
+  // Prefetch won't occur if the host is not healthy.
+  host_->healthFlagSet(Upstream::Host::HealthFlag::DEGRADED_EDS_HEALTH);
+  EXPECT_FALSE(pool_.maybePrefetch(1));
```

Comment:
I think that a prefetch ratio of 1 won't result in prefetch.  Could you set the global prefetch ratio argument to 1.1 or 2?